frontend/catalyst/device/qjit_device.py

```diff
@@ -396,47 +405,48 @@ def preprocess(
 
         return program, config
 
-    def _measurement_transform_program(self):
+    def _measurement_transform_program(self, capabilities=None):
 
+        capabilities = capabilities or self.capabilities
```

Comment:
```suggestion
    def _measurement_transform_program(self, capabilities=None):
        # in case of dynamically supplied shots
        capabilities = capabilities or self.capabilities
```